### ISSUE TITLE
Trying to use functionaljava for android development, please help

### ISSUE THREAD

**Issue by Arjuna144** [functionaljava/functionaljava#311]
Hello, I'm having issues trying to use functionaljava.

I am not an experienced java/android developer and I don't understand how to add functionaljava to my android project. 
This is my grade file:
```
buildscript {
    repositories {
        mavenCentral()
    }

    dependencies {
        classpath 'me.tatarka:gradle-retrolambda:3.4.0'
    }
}

// Required because retrolambda is on maven central
repositories {
    mavenCentral()
}

apply plugin: 'com.android.application' //or apply plugin: 'java'
apply plugin: 'me.tatarka.retrolambda'

android {
    compileSdkVersion 25
    buildToolsVersion "25.0.1"
    defaultConfig {
        applicationId "de.threecoders.hangapp"
        minSdkVersion 15
        targetSdkVersion 25
        versionCode 1
        versionName "1.0"
        testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
    }
    buildTypes {
        release {
            minifyEnabled false
            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
        }
    }
}

dependencies {
    compile fileTree(dir: 'libs', include: ['*.jar'])
    androidTestCompile('com.android.support.test.espresso:espresso-core:2.2.2', {
        exclude group: 'com.android.support', module: 'support-annotations'
    })
    compile "org.functionaljava:functionaljava:4.6"
    compile "org.functionaljava:functionaljava-java-core:4.6"
    compile 'com.android.support:appcompat-v7:25.0.1'
    testCompile 'junit:junit:4.12'
}
```

The IDE (IntelliJ) seems to eat that, since it code-completes and typechecks all the stuff, I'm using from functionaljava. But if I try to run on an android emulator I get this error:
```
Error:Error converting bytecode to dex:
Cause: Dex cannot parse version 52 byte code.
This is caused by library dependencies that have been compiled using Java 8 or above.
If you are using the 'java' gradle plugin in a library submodule add 
targetCompatibility = '1.7'
sourceCompatibility = '1.7'
to that submodule's build.gradle file.
```

Please help me!

**Comment by dobesv** [comment]
It looks like Android doesn't work with Java 8 compiled classes.

I'm not sure off the top of my head whether there's an easy way to get Java
7 compatible class files for the latest functionaljava or not.  You might
need to use a version of functionaljava prior to 4.0.

**Comment by gliptak** [comment]
https://github.com/functionaljava/functionaljava/issues/265
https://github.com/functionaljava/functionaljava/pull/302

**Comment by jbgi** [comment]
FunctionalJava published jar should have been built targeting java 1.6 (through retrolambda), except for the `java8` module.
Unfortunately it looks like the `java-core` module was also published targeting 1.8. This need to be fixed for the upcoming release.
In the meantime, you need to leave out the dependency on `functionaljava-java-core`, and, if needed, copy the source of the only class of that module into your project.

thanks for reporting the problem!

**Comment by jbgi** [comment]
fixed by 3dcdf401182899611608537dd9862f06eed18062
